include/SFML/System/FileInputStream.hpp

```diff
@@ -143,6 +137,12 @@ class SFML_SYSTEM_API FileInputStream : public InputStream
     std::optional<std::size_t> getSize() override;
 
 private:
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    ////////////////////////////////////////////////////////////
+    FileInputStream();
```

Comment:
When constructing a `FileInputStream`, sometimes you construct from an `std::unique_ptr<priv::ResourceStream>` and sometimes from a `std::unique_ptr<std::FILE, FileCloser>`. I suppose we could add two constructors to handle both cases for now chose the lazy solution. Open to to changing if someone feels like have both of those constructors is the right call.

Reply:
Unfortunately I feel like having those constructors is the right call, I'd like to keep empty states unrepresentable even in our own implementation, and to be consistent with the great work we've done so far.

It seems like you should get away with:

```cpp
// FileInputStream.hpp

explicit FileInputStream(std::unique_ptr<std::FILE, FileCloser>&& file);

#ifdef SFML_SYSTEM_ANDROID
explicit FileInputStream(std::unique_ptr<priv::ResourceStream>&& androidFile);
#endif
```

```cpp
// FileInputStream.cpp

FileInputStream::FileInputStream(std::unique_ptr<std::FILE, FileCloser>&& file)
    : m_file(std::move(file))
{
}

#ifdef SFML_SYSTEM_ANDROID
FileInputStream::FileInputStream(std::unique_ptr<priv::ResourceStream>&& androidFile)
    : m_androidFile(std::move(androidFile))
{
}
#endif
```

Which I don't think it's too bad :)